Automattic/wp-calypso
# LOTS: Correct CTA URL for personal tier themes

Related to #85728 which is the equivalent for LITS

## Proposed Changes

When tiered themes are available, `getThemeType` is no longer sufficient to find the tier of the theme - it treats all themes with a `pub/` prefix as free themes. With this change, theme tiers is treated as the source of truth, and mapped to the values expected in onboarding.

Onboarding has been updated so that when starting from the logged-out theme showcase and choosing a personal-tiered theme, it will show all plans except the free plan. This is analagous to how premium themes show all plans except free and personal.

## Testing Instructions

 1. Apply patch / calypso live link
 2. Go to `/themes?flags=themes/tiers` in an incognito window
 3. Click on each type of theme - free, personal (e.g. epi), premium, Woo, Marketplace, Community (need to search first)
 4. Compare the "Pick this design" URLs to those of production - except for personal themes they should be the same
 5. Follow the "Pick this design" URLs, and go through the site creation process, you should see that everything works as on production except that for Personal plans the plan selector will not show the free plan.

<img width="1441" alt="Screenshot 2024-01-05 at 14 00 58" src="https://github.com/Automattic/wp-calypso/assets/93301/91a4e490-5560-488e-ad26-765a06696ffe">


## Pre-merge Checklist
